### DISCUSSION TITLE
Compile error - clang: error: linker command failed with exit code 1 (use -v to see invocation)

### DISCUSSION THREAD

**Discussion by clearmindful** [microsoft/vscode-cpptools#7594] [Q&A]
Hi 

New to C++ and Visual Studio Code. I had followed some videos in getting the Visual Studio Code to work on macOS Mojave 10.14.6 but it just doesnt work ... Perhaps is something trivial so forgive my naivness; it will be wonderful to get someone professional to help. 

With appreciation
<img width="1246" alt="Screenshot 2021-05-26 at 8 23 47 AM" src="https://user-images.githubusercontent.com/42393817/119606750-c5542880-bdfb-11eb-976c-e7352c99da1a.png">
<img width="939" alt="Screenshot 2021-05-26 at 8 23 57 AM" src="https://user-images.githubusercontent.com/42393817/119606761-c9804600-bdfb-11eb-9b1e-e7e669bc33a8.png">

**Comment by sean-mcmanus** [comment]
You're only compiling ClassesAndObjects.cpp, but that code references Person::getName() which appears to be implemented in Person.cpp, which also needs to be compiled, and the resulting object file linked with your executable.

**Comment by clearmindful** [comment]
Thanks for the update. I believe that i should have created all of these
files under project ( or workspace - again forgive me I am not sure how
that works) and should run it all together. However I dont know how to link
all these files under the Visual Studio.. Can you advice?

**Reply by sean-mcmanus** [reply]
Compiling file.c will create file.o, and you can link to other file.o via `-o file.o`.

**Comment by michelleangela** [comment]
@clearmindful 
Here are some documentations about compiling and linking with g++ that might provide more information about compiling and linking.

Note: these documentations/notes appear to be from some college course.
https://www3.ntu.edu.sg/home/ehchua/programming/cpp/gcc_make.html
https://courses.cs.washington.edu/courses/cse373/99au/unix/g++.html

**Comment by clearmindful** [comment]
Hi

Could I kindly ask you to recommend the best c++ course that will
beneficial for me? Don’t get me wrong Udemy courses are very good, but if
there is a course with an instructor that delivers best result in real life
projects and what happens in industry these days, I will go for that in a
blink of an eye.

Thanks 🙏 🌟

**Comment by michelleangela** [comment]
@clearmindful 

>a course with an instructor that delivers best result in real life
projects and what happens in industry these days

I don't know of any C++ course that can really teach and prepare people for this. Also real world experiences can be unique to the specific projects and scenarios (where is the software being used: a home device, a phone, or a vehicle?). From my personal experience developers mostly learn and grow their skills on the job. Usually, as someone takes on new projects then they would need to research and learn the new technologies (if any) that would meet a project's requirements. So what developers need to learn for a project sometimes depends on what the project requires.

In terms of just learning a coding language so you can use it to create software, there are plenty of online course that teach fundamentals to get you started. The online C++ courses I've seen on www.pluralsight.com teach C++ fundamentals, its syntax, and software design patterns usage. There could be more that I haven't seen as well.
